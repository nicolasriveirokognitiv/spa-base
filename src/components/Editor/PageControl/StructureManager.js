import React, { useState } from 'react';
import BlockList from './BlockList/BlockList';
import Item from './BlockList/Item';
import PropTypes from 'prop-types';

function StructureManager({ page: { rows } }) {
    const [Block, setBlock] = useState(null);

    const handleBlockClick = (isOpen, block) => setBlock(isOpen ? null : block);

    return Block
        ? (
            <>
                <Item isOpen onClick={handleBlockClick} block={Block} />
                <Block.Toolset {...Block.properties} />
            </>
        )
        : <BlockList rows={rows} onClick={handleBlockClick} />;
}

StructureManager.propTypes = { page: PropTypes.shape({ rows: PropTypes.arrayOf(PropTypes.shape({})) }).isRequired };

export default StructureManager;
