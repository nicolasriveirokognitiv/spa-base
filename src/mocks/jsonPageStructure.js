export default {
    config: {
        title             : 'SPA Base',
    },
    rows : [
        {
            columns: [
                {
                    width  : '100',
                    blocks : [
                        {
                            id       : 'NgVHq6',
                            type     : 'Notification',
                            category : 'Basic',
                            config   : {},
                            design   : {
                                border        : {},
                                borderRadius  : 'border-radius-slightly',
                                textAlignment : 'flex-left',
                            },
                            data: {
                                description : 'This is a notification for you to be notified',
                                title       : 'Notification Title',
                            },
                        },
                    ],
                },
            ],
        },
        {
            columns: [
                {
                    width  : '100',
                    blocks : [
                        {
                            id       : 'xaF0kvS',
                            type     : 'Notification',
                            category : 'Headers',
                            config   : { title: 'Dummy' },
                            design   : { size: 'Medium' },
                            data     : {
                                description : 'This is a notification for you to be notified',
                                title       : 'Notification Title',
                            },
                        },
                    ],
                },
            ],
        },
        {
            columns: [
                {
                    width  : '100',
                    blocks : [
                        {
                            id       : 'jf3FD3',
                            type     : 'Card',
                            category : 'Headers',
                            config   : { title: 'Dummy' },
                            design   : { size: 'Medium' },
                            data     : {
                                header      : 'Block title',
                                title       : 'Best travel destination ever',
                                description : 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. In mollis sem sit amet urna egestas, eget dapibus tellus malesuada. Ut eu elementum lacus. Ut eu elementum lacus.',
                                buttonLabel : 'Sample Text',
                            },
                        },
                    ],
                },
            ],
        },
    ],
};
