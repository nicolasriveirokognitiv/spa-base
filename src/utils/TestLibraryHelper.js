/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import initStore from 'State/store';

const store = initStore();

const renderWithRedux = children => render(
    <Provider store={store}>
        {children}
    </Provider>,
);

export * from '@testing-library/react';
export { renderWithRedux, render };
