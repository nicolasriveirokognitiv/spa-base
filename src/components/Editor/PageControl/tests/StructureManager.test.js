import React from 'react';
import StructureManager from '../StructureManager';
import { render } from 'Utils/TestLibraryHelper';
import jsonPageStructure from '../../../../mocks/jsonPageStructure';

describe('StructureManager', () => {
    let props;
    let container;

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
        props = {
            page             : jsonPageStructure,
            handleBlockClick : jest.fn(),
        };
    });

    test('Should render correctly', () => {
        expect(<StructureManager {...props} />).toBeDefined();
    });

    test('Should match snapshot', async () => {
        const { asFragment } = render(<StructureManager {...props} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
