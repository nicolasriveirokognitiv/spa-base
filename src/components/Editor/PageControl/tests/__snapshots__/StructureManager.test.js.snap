// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StructureManager Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="page-control-item"
  >
    <div
      class="page-control-item-header"
    >
      <span
        class="page-control-item-header-icon bm-icon-message-alert"
      />
      <span
        class="page-control-item-header-name"
      >
        Notification
      </span>
    </div>
  </div>
  <div
    class="page-control-item"
  >
    <div
      class="page-control-item-header"
    >
      <span
        class="page-control-item-header-icon bm-icon-message-alert"
      />
      <span
        class="page-control-item-header-name"
      >
        Notification
      </span>
    </div>
  </div>
  <div
    class="page-control-item"
  >
    <div
      class="page-control-item-header"
    >
      <span
        class="page-control-item-header-icon bm-icon-image-with-text"
      />
      <span
        class="page-control-item-header-name"
      >
        Image with text
      </span>
    </div>
  </div>
</DocumentFragment>
`;
